### DISCUSSION TITLE
close action modal after validation

### DISCUSSION THREAD

**Discussion by mikedev20** [filamentphp/filament#13425] [Help]
### Package

Actions

### Package Version

v3.2

### How can we help you?

hello, in my action I put a modal and I would like that at its submission I close the modal and I launch the validation of fields
I get to the catch but it doesn't launch the close-modal event.
In my case, it launches the validation while the modal is still open.
If I remove the validate, it closes the modal.
Thank you very much for your help

   return Action::make('preview')
            ->action(function(Component $livewire, Action $action, Request $request){
           try {
                    $livewire->validate();
                } catch (Throwable $e) {
                    info("i'm here");
                    $action->dispatch('close-modal');
                    throw $e;
                }
            })
            ->modalContent(function ($livewire, Action $action) {
                $annonce = new Annonce();
                $annonce->fill($livewire->data);
                return view(
                    'filament.actions.annonce-preview',
                    ['annonce' => $annonce],
                );
            });

**Comment by tnt-ylli** [comment]
anything on this ?
